### PR TITLE
Replace trigger_error with wp_trigger_error

### PR DESCRIPTION
**Pull request by prasadkarmalkar**
In this PR I have replaced trigger_error with wp_trigger_error and for the function name first parameter in wp_trigger_error I have used __METHOD__ and __FUNCTION__ based on either this is in function or class

Trac ticket:  https://core.trac.wordpress.org/ticket/59652

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by Rajinsharwar — CHANGES_REQUESTED** [review]
Can you please strip all tags before displaying the error?

**Comment by thelovekesh** [comment]
Hi @Rajinsharwar!

Is there a specific reason to strip HTML tags from triggered error messages? I believe the HTML tags are expected, allowing the front end to render the error messages makeup(e.g. links) appropriately for the user.

Regarding the failed tests, `wp_trigger_error()` includes the method or function name in the error message, which is causing the tests to fail.

https://github.com/WordPress/wordpress-develop/blob/f1c0e5856e71f1431e1cd1ef9d6775bb8495b532/src/wp-includes/functions.php#L6065

_Please note, how function/method name is being appended in the front of error message._

I believe the test cases need to be updated to account for the `__FUNCTION__` name prefix if it is being passed to `wp_trigger_error()`.

File: tests/phpunit/tests/filesystem/wpOpcacheInvalidateDirectory.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ public static function set_up_before_class() {
 	public function test_should_trigger_error_with_invalid_dir( $dir ) {
 		$this->expectError();
 		$this->expectErrorMessage(
-			'<code>wp_opcache_invalidate_directory()</code>',
+			'wp_opcache_invalidate_directory(): wp_opcache_invalidate_directory() expects a non-empty string.',
```
Comment (thelovekesh) [inline tests/phpunit/tests/filesystem/wpOpcacheInvalidateDirectory.php]: `<code>...</code>` is expected here and shouldn't get stripped. But the tests are passing due to a bug in `wp_trigger_error()` HTML escaping. I have filed https://core.trac.wordpress.org/ticket/61318 to address the bug.

Once https://github.com/WordPress/wordpress-develop/pull/6671 is committed in trunk, this needs to be updated to 

```suggestion
			'wp_opcache_invalidate_directory(): <code>wp_opcache_invalidate_directory()</code> expects a non-empty string.',
```

Comment (prasadkarmalkar) [reply tests/phpunit/tests/filesystem/wpOpcacheInvalidateDirectory.php]: Thank you @thelovekesh, I will update this once the escaping issue is resolved 🗒️

Comment (thelovekesh) [reply tests/phpunit/tests/filesystem/wpOpcacheInvalidateDirectory.php]: @prasadkarmalkar https://github.com/WordPress/wordpress-develop/pull/6671 is merged into the trunk. Please feel free to make your changes accordingly.

Comment (prasadkarmalkar) [reply tests/phpunit/tests/filesystem/wpOpcacheInvalidateDirectory.php]: Updated the test cases

**Comment by hellofromtonya** [comment]
Hello and thanks for working on this effort 👋  Some of the changes are in external packages / libraries that are maintained outside of Core, e.g. Requests, PHPMailer, SimplePie, etc. 

For this first round of replacements (especially this late in the beta cycle), I'm thinking to only replace within Core's maintained codebase. For these external libraries, discussion can happen whether in their repo (such as with Requests) or within 6.7+ trac ticket.

I'll push a change to this PR shortly that reverts the changes to external libraries. And then review the remaining changes for 6.6 commit consideration.

**Comment by hellofromtonya** [comment]
I did a force push because it easiest and most clean way for me to restore the changes in external libraries from the original first commit https://github.com/WordPress/wordpress-develop/pull/6621/commits/c99f02736e745cddc5e23cea572a2d8a542d5d2b. Also rebased it on top of the current WP `trunk` just in case.

File: src/wp-includes/block-template-utils.php [hunk 2]
```diff
@@ -252,7 +252,7 @@ static function ( $item ) {
 		$type,
 		WP_TEMPLATE_PART_AREA_UNCATEGORIZED
 	);
-	trigger_error( $warning_message, E_USER_NOTICE );
+	wp_trigger_error( __FUNCTION__, $warning_message, E_USER_NOTICE );
```
Comment (hellofromtonya) [inline src/wp-includes/block-template-utils.php]: `E_USER_NOTICE` is not longer needed as it's the default.
```suggestion
	wp_trigger_error( __FUNCTION__, $warning_message );
```

File: src/wp-includes/functions.php [hunk 3]
```diff
@@ -8757,7 +8759,8 @@ function recurse_dirsize( $directory, $exclude = null, $max_execution_time = nul
  */
 function clean_dirsize_cache( $path ) {
 	if ( ! is_string( $path ) || empty( $path ) ) {
-		trigger_error(
+		wp_trigger_error(
+			__FUNCTION__,
```
Comment (hellofromtonya) [inline src/wp-includes/functions.php]: The function's name is already incorporated into the translated version of the error message.

```suggestion
			'',
```

File: src/wp-includes/rss.php [hunk 4]
```diff
@@ -63,7 +63,8 @@ function __construct( $source ) {
 		# Check if PHP xml isn't compiled
 		#
 		if ( ! function_exists('xml_parser_create') ) {
-			return trigger_error( "PHP's XML extension is not available. Please contact your hosting provider to enable PHP's XML extension." );
+			wp_trigger_error( __METHOD__, "PHP's XML extension is not available. Please contact your hosting provider to enable PHP's XML extension." );
```
Comment (hellofromtonya) [inline src/wp-includes/rss.php]: Thinking the constructor method within this error message is not adding value and may confuse.
```suggestion
			wp_trigger_error( '', "PHP's XML extension is not available. Please contact your hosting provider to enable PHP's XML extension." );
```

File: src/wp-includes/rss.php [hunk 5]
```diff
@@ -387,7 +388,7 @@ function map_attrs($k, $v) {
 
 	function error( $errormsg, $lvl = E_USER_WARNING ) {
 		if ( MAGPIE_DEBUG ) {
-			trigger_error( $errormsg, $lvl);
+			wp_trigger_error( __METHOD__, $errormsg, $lvl);
```
Comment (hellofromtonya) [inline src/wp-includes/rss.php]: I'm not sure if adding the method to the error message will be of value in RSS. For now, suggest removing it.

```suggestion
			wp_trigger_error('', $errormsg, $lvl);
```

File: src/wp-includes/rss.php [hunk 6]
```diff
@@ -823,7 +824,7 @@ function file_name ($url) {
 	function error ($errormsg, $lvl=E_USER_WARNING) {
 		$this->ERROR = $errormsg;
 		if ( MAGPIE_DEBUG ) {
-			trigger_error( $errormsg, $lvl);
+			wp_trigger_error( __METHOD__, $errormsg, $lvl);
```
Comment (hellofromtonya) [inline src/wp-includes/rss.php]: Same here.
```suggestion
			wp_trigger_error( '', $errormsg, $lvl);
```

File: src/wp-includes/theme.php [hunk 7]
```diff
@@ -514,7 +514,7 @@ function search_theme_directories( $force = false ) {
 		// Start with directories in the root of the active theme directory.
 		$dirs = @ scandir( $theme_root );
 		if ( ! $dirs ) {
-			trigger_error( "$theme_root is not readable", E_USER_NOTICE );
+			wp_trigger_error( __FUNCTION__, "$theme_root is not readable", E_USER_NOTICE );
```
Comment (hellofromtonya) [inline src/wp-includes/theme.php]: The `E_USER_NOTICE` is the default. Can safely remove it.

```suggestion
			wp_trigger_error( __FUNCTION__, "$theme_root is not readable" );
```

File: src/wp-includes/theme.php [hunk 8]
```diff
@@ -538,7 +538,7 @@ function search_theme_directories( $force = false ) {
 				 */
 				$sub_dirs = @ scandir( $theme_root . '/' . $dir );
 				if ( ! $sub_dirs ) {
-					trigger_error( "$theme_root/$dir is not readable", E_USER_NOTICE );
+					wp_trigger_error( __FUNCTION__, "$theme_root/$dir is not readable", E_USER_NOTICE );
```
Comment (hellofromtonya) [inline src/wp-includes/theme.php]:
```suggestion
					wp_trigger_error( __FUNCTION__, "$theme_root/$dir is not readable" );
```

File: src/wp-admin/includes/file.php [hunk 9]
```diff
@@ -2795,7 +2795,7 @@ function wp_opcache_invalidate_directory( $dir ) {
 				__( '%s expects a non-empty string.' ),
 				'<code>wp_opcache_invalidate_directory()</code>'
 			);
-			trigger_error( $error_message );
+			wp_trigger_error( __FUNCTION__, $error_message );
```
Comment (hellofromtonya) [inline src/wp-admin/includes/file.php]: The function's name is already in the error message. It's redundant to add it again.
```suggestion
			wp_trigger_error( '', $error_message );
```

File: tests/phpunit/tests/filesystem/wpOpcacheInvalidateDirectory.php [hunk 10]
```diff
@@ -37,7 +37,7 @@ public static function set_up_before_class() {
 	public function test_should_trigger_error_with_invalid_dir( $dir ) {
 		$this->expectError();
 		$this->expectErrorMessage(
-			'<code>wp_opcache_invalidate_directory()</code>',
+			'wp_opcache_invalidate_directory(): <code>wp_opcache_invalidate_directory()</code> expects a non-empty string.',
```
Comment (hellofromtonya) [inline tests/phpunit/tests/filesystem/wpOpcacheInvalidateDirectory.php]: As suggested above, removing the redundant function name.

```suggestion
			'<code>wp_opcache_invalidate_directory()</code> expects a non-empty string.',
```

File: src/wp-includes/formatting.php [hunk 11]
```diff
@@ -5518,10 +5518,11 @@ function wp_strip_all_tags( $text, $remove_breaks = false ) {
 	if ( ! is_scalar( $text ) ) {
 		/*
 		 * To maintain consistency with pre-PHP 8 error levels,
-		 * trigger_error() is used to trigger an E_USER_WARNING,
+		 * wp_trigger_error() is used to trigger an E_USER_WARNING,
 		 * rather than _doing_it_wrong(), which triggers an E_USER_NOTICE.
 		 */
-		trigger_error(
+		wp_trigger_error(
+			__FUNCTION__,
```
Comment (hellofromtonya) [inline src/wp-includes/formatting.php]: The function's is already in the error message. Can safely be removed here.

```suggestion
			'',
```

File: tests/phpunit/tests/functions/cleanDirsizeCache.php [hunk 12]
```diff
@@ -35,19 +35,19 @@ public function data_clean_dirsize_cache_with_invalid_inputs() {
 		return array(
 			'null'         => array(
 				'path'             => null,
-				'expected_message' => '<code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>NULL</code>.',
+				'expected_message' => 'clean_dirsize_cache(): <code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>NULL</code>.',
```
Comment (hellofromtonya) [inline tests/phpunit/tests/functions/cleanDirsizeCache.php]: As suggested above, adding the function name is redundant as the error message already has it built in.

```suggestion
				'expected_message' => '<code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>NULL</code>.',
```

File: tests/phpunit/tests/functions/cleanDirsizeCache.php [hunk 13]
```diff
@@ -35,19 +35,19 @@ public function data_clean_dirsize_cache_with_invalid_inputs() {
...
 			),
 			'bool false'   => array(
 				'path'             => false,
-				'expected_message' => '<code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>boolean</code>.',
+				'expected_message' => 'clean_dirsize_cache(): <code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>boolean</code>.',
```
Comment (hellofromtonya) [inline tests/phpunit/tests/functions/cleanDirsizeCache.php]:
```suggestion
				'expected_message' => '<code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>boolean</code>.',
```

File: tests/phpunit/tests/functions/cleanDirsizeCache.php [hunk 14]
```diff
@@ -35,19 +35,19 @@ public function data_clean_dirsize_cache_with_invalid_inputs() {
...
 			),
 			'empty string' => array(
 				'path'             => '',
-				'expected_message' => '<code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>string</code>.',
+				'expected_message' => 'clean_dirsize_cache(): <code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>string</code>.',
```
Comment (hellofromtonya) [inline tests/phpunit/tests/functions/cleanDirsizeCache.php]:
```suggestion
				'expected_message' => '<code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>string</code>.',
```

File: tests/phpunit/tests/functions/cleanDirsizeCache.php [hunk 15]
```diff
@@ -35,19 +35,19 @@ public function data_clean_dirsize_cache_with_invalid_inputs() {
...
 			),
 			'array'        => array(
 				'path'             => array( '.', './second/path/' ),
-				'expected_message' => '<code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>array</code>.',
+				'expected_message' => 'clean_dirsize_cache(): <code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>array</code>.',
```
Comment (hellofromtonya) [inline tests/phpunit/tests/functions/cleanDirsizeCache.php]:
```suggestion
				'expected_message' => '<code>clean_dirsize_cache()</code> only accepts a non-empty path string, received <code>array</code>.',
```

File: src/wp-admin/includes/plugin-install.php [hunk 16]
```diff
@@ -174,7 +174,8 @@ function plugins_api( $action, $args = array() ) {
 
 		if ( $ssl && is_wp_error( $request ) ) {
 			if ( ! wp_is_json_request() ) {
-				trigger_error(
+				wp_trigger_error(
+					__FUNCTION__,
```
Comment (hellofromtonya) [inline src/wp-admin/includes/plugin-install.php:178]:
```suggestion
					'',
```

File: src/wp-admin/includes/theme.php [hunk 17]
```diff
@@ -568,7 +568,8 @@ function themes_api( $action, $args = array() ) {
 
 		if ( $ssl && is_wp_error( $request ) ) {
 			if ( ! wp_doing_ajax() ) {
-				trigger_error(
+				wp_trigger_error(
+					__FUNCTION__,
```
Comment (hellofromtonya) [inline src/wp-admin/includes/theme.php:572]:
```suggestion
					'',
```

**Review by hellofromtonya — APPROVED** [review]
Ready for commit 👍

**Comment by hellofromtonya** [comment]
Thank you everyone for your contributions 🌟 

Committed via https://core.trac.wordpress.org/changeset/58409 to ship in 6.6.

**Pull request closed without merging**